### ISSUE TITLE
Export/import

### ISSUE THREAD

**Issue by impworks** [impworks/bonsai#11]
It would be nice to have import/export to popular genealogy data formats, such as GEDCOM.

**Comment by Astromis** [comment]
@impworks привет)
Есть ли шансы, что эта фича поедет в работу в обозримом будущем?

**Comment by impworks** [comment]
@Astromis, здравствуйте! Фича была записана в очень абстрактном виде, пока даже исследовать не начинал. Расскажите, пожалуйста, что конкретно для вас важно - импорт, экспорт, какие данные, какой порядок объема и т.д.

**Comment by Astromis** [comment]
@impworks для меня важен экспорт в первую очередь. Очень нравится ваше приложение, как и писал здесь раньше, но каждый раз останавливает то, что если вдруг поддержка прекратится, то данные нельзя будет вытащить. Ну точнее можно, но будет сопряжен с очень сильной болью, потому что я не net разработчик)

Касательно каких данных: условно вся информация в тексте: кто, когда родился, кому кем приходится и т.д. В целом, всё, что покрывается форматом gedcom. Я полагаю, что у вас реализовано гораздо больше, но я думаю, у вас нет ничего такого, чегоня нельзя привести к формату гедкома.

Я готов оказать посильную помочь, вот только, как я сказал, я не .net-разраб)

**Comment by Astromis** [comment]
@impworks позвольте спросить, я же правильно понимаю, что описание тех трех таблиц находится в соответствующих .cs файлах?

**Comment by impworks** [comment]
Правильно, но там много неинтересного. Напримео, все что имеет тип `ICollection<...>` - это данные из других таблиц, подбираемые по FK. А `LivingBeingOverview` - кешированая информация из фактов для быстрого отображения, ценности не представляет.

Сами факты хранятся в виде сериализованного JSON в поле `Page.Facts`. Существующие типы фактов и конкретный формат каждого из них можно посмотреть, начиная с [FactDefinitions.cs](https://github.com/impworks/bonsai/blob/master/src/Bonsai/Code/DomainModel/Facts/FactDefinitions.cs) - даже если вы не C#-разработчик, но (судя по проектам) знаете C++, никакой хитрой логики в них нет, только декларативное описание.

**Comment by impworks** [comment]
Пожалуйста, сделайте в "Дереве жизни" небольшой проект-пример, допустим 5 человек с различными данными, выгрузите его и приложите сюда. Так можно будет хотя бы проэкспериментировать с существующими библиотеками и оценить, насколько они справляются.

Особо интересно, что будет с фотками. Насколько я понимаю, GEDCOM их не хранит внутри и будет содержать пути, т.к. помимо самого файла нужен будет архив с остальными файлами

**Comment by XBaX01** [comment]
> Пожалуйста, сделайте в "Дереве жизни" небольшой проект-пример, допустим 5 человек с различными данными, выгрузите его и приложите сюда. Так можно будет хотя бы проэкспериментировать с существующими библиотеками и оценить, насколько они справляются.
> 
> Особо интересно, что будет с фотками. Насколько я понимаю, GEDCOM их не хранит внутри и будет содержать пути, т.к. помимо самого файла нужен будет архив с остальными файлами

[Архив.zip](https://github.com/user-attachments/files/21426706/default.zip)

Вот небольшой кусочек экспортированный в формате. Фото он положил в папку рядом.
